cpu/cc2538/periph/gpio.c

```diff
@@ -26,19 +26,27 @@
 #include "periph/gpio.h"
 
 #define GPIO_MASK           (0xfffff000)
-#define PORTNUM_MASK        (0x00007000)
+#define PORTNUM_MASK        (0x00003000)
 #define PORTNUM_SHIFT       (12U)
 #define PIN_MASK            (0x00000007)
 #define MODE_NOTSUP         (0xff)
 
 static inline cc2538_gpio_t *gpio(gpio_t pin)
 {
-    return (cc2538_gpio_t *)(pin & GPIO_MASK);
+    if(((uint32_t)pin &GPIO_MASK) == 0){
```

Comment:
Right now there are two types of pin encoding. The one provided by cpu/cc2538/include/cc2538-gpio.h and the one which is used here, provided by the RIOT gpio interface. The one in cc2538-gpio.h is also used for the pin name defines, which are then used for configuring boards, such as the Re-Mote. Because one encoding does not translate well into the other, SPI Pins for example, are not correctly initialized. The error can be reproduced by simply initializing pins through their pin-names, i.e. GPIO_PD4 and then setting or reading them. The result will be, that nothing happens, as the pin will not be propably addressed. This workaround just checks, whether the old or the new encoding is used. And if the old encoding is detected, it then translates it according to the new gpio interface pin encoding. As I mentioned, this should only be a temporary workaround. The fix would be to adjust cc2538-gpio.h pin encoding and all its functions. I have stated one example in the issue, linked to this pull request. I hope I was able to answer your questions adequately.
Cheers